### PR TITLE
Changes for fmi3 refactor in fmi.jl

### PR DESCRIPTION
**Pull request by JoKircher**
to get the tests in FMI.jl to work these minor PR has to be merged first.

### CODE REVIEW

File: src/FMI3/ext.jl [hunk 1]
```diff
@@ -1107,49 +1107,46 @@ function fmi3Get!(inst::FMU3Instance, vrs::fmi3ValueReferenceFormat, dstArray::A
...
             dstArray[i] = fmi3GetBoolean(inst, vr)
-        elseif mv.datatype.datatype == fmi3String 
+        elseif isa(mv, FMICore.mvString)
             #@assert isa(dstArray[i], String) "fmi3Get!(...): Unknown data type for value reference `$(vr)` at index $(i), should be `String`, is `$(typeof(dstArray[i]))`."
             dstArray[i] = fmi3GetString(inst, vr)
-        elseif mv.datatype.datatype == fmi3String 
-            #@assert isa(dstArray[i], String) "fmi3Get!(...): Unknown data type for value reference `$(vr)` at index $(i), should be `String`, is `$(typeof(dstArray[i]))`."
-            dstArray[i] = fmi3GetString(inst, vr)
-        elseif mv.datatype.datatype == fmi3Binary 
+        elseif tisa(mv, FMICore.mvBinary)
```
Comment (ThummeTo) [inline src/FMI3/ext.jl:1146]: `tisa`?

File: src/FMI3/c.jl [hunk 2]
```diff
@@ -355,7 +355,7 @@ function fmi3ExitInitializationMode(c::FMU3Instance)
         elseif fmi3IsScheduledExecution(c.fmu)
             c.state = fmi3InstanceStateClockActivationMode
         else
-            c.state = fmi3InstanceStateEventMode
+            c.state = fmi3InstanceStateContinuousTimeMode
```
Comment (ThummeTo) [inline src/FMI3/c.jl:358]: compliant to state machine in FMI3-sepc?

**Comment by ThummeTo** [comment]
Please add project.toml version inc (minor change, last digit).
Thanks!

File: src/FMI3/c.jl (hunk 2 above)
Comment (JoKircher) [reply src/FMI3/c.jl]: nope, artifact of a test https://github.com/ThummeTo/FMIImport.jl/pull/70/commits/d44b18dde7be53f4325643f290ced1b8651ca530

**Comment by JoKircher** [comment]
fixed in https://github.com/ThummeTo/FMIImport.jl/pull/70/commits/d44b18dde7be53f4325643f290ced1b8651ca530

**Pull request merged**